phoenix-core/src/main/java/org/apache/phoenix/execute/PhoenixTxIndexMutationGenerator.java

```diff
@@ -313,7 +326,18 @@ public int compare(Cell o1, Cell o2) {
     private void generateDeletes(PhoenixIndexMetaData indexMetaData,
             Collection<Pair<Mutation, byte[]>> indexUpdates,
             byte[] attribValue, TxTableState state) throws IOException {
-        Iterable<IndexUpdate> deletes = codec.getIndexDeletes(state, indexMetaData);
+        byte[] regionStartKey = this.regionStartKey;
```

Comment:
Just for my understanding, why do we now need the start and stop key to generate the index deletes and upserts?